### PR TITLE
Fix fasta input

### PR DESCRIPTION
**Pull request by jtangrot**
This PR addresses #542. A module was added to change tabs to spaces in description part of sequence names when using fasta input. The descriptions are kept as part of the sequence name. Removing the tabs will remove the problem with the sequence on a line of its own.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jtangrot** [comment]
I have added descriptions to some of the file names in the fasta file used for testing. I don't know what else needs changing in the testing?

**Comment by erikrikarddaniel** [comment]
> I have added descriptions to some of the file names in the fasta file used for testing. I don't know what else needs changing in the testing?

Perfect.

**Comment by erikrikarddaniel** [comment]
There are tests failing, so I'm waiting with my review. Need to go on an errand now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#520](https://github.com/nf-core/ampliseq/pull/520),[#530](https://github.com/nf-core/ampliseq/pull/530) - Fix conda packages
 - [#531](https://github.com/nf-core/ampliseq/pull/531) - Update documentation
 - [#535](https://github.com/nf-core/ampliseq/pull/535) - Make sure barrnap runs with fasta input
+- [#](https://github.com/nf-core/ampliseq/pull/) - Adding module to fix header in fasta input if needed
```
Comment (d4straub) [inline CHANGELOG.md:26]:
```suggestion
- [#544](https://github.com/nf-core/ampliseq/pull/544) - Adding module to fix header in fasta input if needed
```

**Review by d4straub — COMMENTED** [review]
Looks good to me, I'll restart tests, maybe the errors are on github (since they seem unrelated to your changes).

**Comment by d4straub** [comment]
There is a reproducible (here in github) error with `-profile test,docker`. The error message is

```
No such variable: info

 -- Check script './workflows/../subworkflows/local/parse_input.nf' at line: 63 or see '.nextflow.log' file for more details
Argument of 'channel.from' method cannot be a channel object — Likely you can replace the use of 'channel.from' with the channel object itself

 -- Check script './workflows/ampliseq.nf' at line: 652 or see '.nextflow.log' file for more details
```
I do not understand whats happening though.

**Comment by erikrikarddaniel** [comment]
> There is a reproducible (here in github) error with `-profile test,docker`. The error message is
> 
> ```
> No such variable: info
> 
>  -- Check script './workflows/../subworkflows/local/parse_input.nf' at line: 63 or see '.nextflow.log' file for more details
> Argument of 'channel.from' method cannot be a channel object — Likely you can replace the use of 'channel.from' with the channel object itself
> 
>  -- Check script './workflows/ampliseq.nf' at line: 652 or see '.nextflow.log' file for more details
> ```
> 
> I do not understand whats happening though.

The error message is clear enough -- and `Channel.from(ch_multiqc_config)` looks like it could be the offender, since it's calling the method with a channel (possibly empty) -- but strange that it only affects docker and shows up in a file that's not touched. It's input can have been touched of course, but still strange that singularity works -- or does it (I don't really understand this test)?

**Comment by d4straub** [comment]
Well, that works for me flawless:
```
nextflow pull jtangrot/ampliseq
nextflow pull jtangrot/ampliseq -r fix_fasta_input
nextflow run jtangrot/ampliseq -r fix_fasta_input -profile test,singularity --outdir results_jtangrot-fix_fasta_input__test
nextflow run jtangrot/ampliseq -r fix_fasta_input -profile test,docker --outdir results_jtangrot-fix_fasta_input__test_docker
```

**Comment by jtangrot** [comment]
Thanks for looking into this. I really have no clue how to fix this...

**Comment by d4straub** [comment]
Maybe bother the github actions experts in the nf-core slack?

**Comment by d4straub** [comment]
The failing test `nf-core CI / Run pipeline with test data (latest-everything) (pull_request)` uses `N E X T F L O W  ~  version 23.02.0-edge`, which was released 3 days ago. I strongly suspect that the nextflow version causes that issue. If that is true, then we can ignore the failing test for that PR. I can check that on Monday.

**Comment by erikrikarddaniel** [comment]
Or maybe fix the problem so it works with the new version? Sooner or later we might have to anyway.

**Comment by erikrikarddaniel** [comment]
Did the test get stuck? I think this looks fine with the change to the channel, but wanted to wait until the test completed.

**Comment by d4straub** [comment]
`NXF_VER=23.02.0-edge nextflow run nf-core/ampliseq -r 2.4.1 -profile test,singularity` leads to the same error, therefore confirmed that its the nextflow version, not this PR.

> Or maybe fix the problem so it works with the new version? Sooner or later we might have to anyway.

Thats an idea, but because it seems to have nothing to do with this PR, I was rather leaning towards fixing this separately. Anyway, the [latest commit up there](https://github.com/nf-core/ampliseq/pull/544/commits/9b4fe25e8563d62435653ce1f3e6cce9cb831537) is only addressing the problem partially I think. It seems we deviate from the [nf-core template 2.7.2](https://github.com/nf-core/tools/blob/99961bedab1518f592668727a4d692c4ddf3c336/nf_core/pipeline-template/workflows/pipeline.nf#L88-L110), i.e. the whole line might have to be removed?
I guess that was my mistake when doing the template update.

**Comment by erikrikarddaniel** [comment]
I'm fine with either approving this latest commit or making a fuller attempt at addressing issues from an upgrade of Nextflow.

File: workflows/ampliseq.nf [hunk 2]
```diff
@@ -647,7 +649,7 @@ workflow AMPLISEQ {
             ch_multiqc_files = ch_multiqc_files.mix(CUTADAPT_WORKFLOW.out.logs.collect{it[1]}.ifEmpty([]))
         }
 
-        ch_multiqc_configs = Channel.from(ch_multiqc_config).mix(ch_multiqc_custom_config).ifEmpty([])
+        ch_multiqc_config = ch_multiqc_config.mix(ch_multiqc_custom_config).ifEmpty([])
```
Comment (d4straub) [inline workflows/ampliseq.nf:653]: I deleted the line locally and tested with `NXF_VER=23.02.0-edge` & `NXF_VER=22.10.1` and it worked fine, so I suggest to delete the line.

```suggestion
```

Comment (erikrikarddaniel) [reply workflows/ampliseq.nf]: Fine. I suppose @jtangrot should commit...

Comment (jtangrot) [reply workflows/ampliseq.nf]: If you are fine with deleting that line, I am :) By the way, I don't think it was really used before, the channel was previously named ch_multiqc_configs (with an 's' at the end) and I cannot find it anywhere else. However, I thought that was a typo, and it runs nicely locally.

Comment (d4straub) [reply workflows/ampliseq.nf]: Sorry, I wanted to save some time and commit already. But now github tests are queuing anyway, so no time benefit ;(

Comment (d4straub) [reply workflows/ampliseq.nf]: > ch_multiqc_configs (with an 's' at the end)

indeed, and I thought it was duplicated, but no, it was even non-functional...

**Review by d4straub — APPROVED** [review]
LGTM!

**Comment by jtangrot** [comment]
Thanks!

**Pull request merged**